Tribler/community/tunnel/hidden_community.py

```diff
@@ -294,17 +374,22 @@ def check_key_request(self, messages):
 
     def on_key_request(self, messages):
         for message in messages:
-            # if we have received this message over a socket, we need to forward it
             if not message.source.startswith(u"circuit_"):
+                # The intropoint receives the message over a socket, and forwards it to the seeder
                 self._logger.debug("On key request: relay key request")
+                cache = self.request_cache.add(KeyRelayCache(self, message.payload.identifier, message.payload.circuit_id))
+                meta = self.get_meta_message(u'key-request')
+                message = meta.impl(distribution=(self.global_time,), payload=(message.payload.circuit_id, cache.number, message.payload.info_hash))
                 relay_circuit = self.intro_point_for[message.payload.info_hash]
-                relay_circuit.tunnel_data(message.candidate.sock_addr, TUNNEL_PREFIX + message.packet)
```

Comment:
I also thought that. But when the key-response reached the intro-point (which should relay the response to the downloader over the originating circuit), it did not use the originating circuit but it sends directly to the downloader. 

Even more: it meant that the sock_addr of the downloader was known by the seeder? Than the whole hidden services would not make sense.